tests/java/junit/DateTest.java

```diff
@@ -0,0 +1,188 @@
+        assertEquals(d.year, args[0]);
+        assertEquals(d.month, args[1]);
+        assertEquals(d.day, args[2]);
+    }
+
+    @Test
+    public void testWeekday() {
+        org.python.Object[] args = { 
+            Int.getInt(2020), // year
+            Int.getInt(10), // month
+            Int.getInt(24), // day, 24 october is an saturday
+        };
+        Date d = new Date(args, Collections.EMPTY_MAP);
+
+        assertEquals(d.weekday(), Int.getInt(5));
+    }
+
+    @Test
+    public void testISOWeekday() {
+        org.python.Object[] args = { 
+            Int.getInt(2020), // year
+            Int.getInt(10), // month
+            Int.getInt(24), // day, 24 october is an saturday
+        };
+        Date d = new Date(args, Collections.EMPTY_MAP);
+
+        assertEquals(d.isoweekday(), Int.getInt(6));
+    }
+
+    @Test(expected = TypeError.class)
+    public void testMissingDay() {
+        org.python.Object[] args = { 
+            Int.getInt(2020), // year
+            Int.getInt(10), // month
+        };
+        Date d = new Date(args, Collections.EMPTY_MAP);
+    }
+
+    @Test(expected = TypeError.class)
+    public void testMissingMonth() {
+        org.python.Object[] args = {
+            Int.getInt(2020), // year
+            Int.getInt(24), // day
+        };
+        Date d = new Date(args, Collections.EMPTY_MAP);
+    }
+
+    @Test(expected = TypeError.class)
+    public void testMissingYear() {
+        org.python.Object[] args = {
+            Int.getInt(10), // month
+            Int.getInt(24), // day
+        };
+        Date d = new Date(args, Collections.EMPTY_MAP);
+    }
+
+    @Test(expected = ValueError.class)
+    public void testMonthOutOfRange() {
+        org.python.Object[] args = {
+            Int.getInt(2020),
+            Int.getInt(13), // month
+            Int.getInt(24), // day
+        };
+        Date d = new Date(args, Collections.EMPTY_MAP);
+    }
+
+    @Test(expected = ValueError.class)
+    public void testMonthZero() {
+        org.python.Object[] args = {
+            Int.getInt(2020),
+            Int.getInt(0), // month
+            Int.getInt(24), // day
+        };
+        Date d = new Date(args, Collections.EMPTY_MAP);
+    }
+
+    @Test(expected = ValueError.class)
+    public void testDayOutOfRange() {
+        org.python.Object[] args = {
+            Int.getInt(2020),
+            Int.getInt(6), // month
+            Int.getInt(32), // day
+        };
+        Date d = new Date(args, Collections.EMPTY_MAP);
+    }
+
+    @Test(expected = ValueError.class)
+    public void testDayZero() {
+        org.python.Object[] args = {
+            Int.getInt(2020),
+            Int.getInt(6), // month
+            Int.getInt(0), // day
+        };
+        Date d = new Date(args, Collections.EMPTY_MAP);
+    }
+
+    @Test
+    public void testKwargs() {
+        Map kwargs = new HashMap<java.lang.String, org.python.Object>();
```

Comment:
```suggestion
        Map<java.lang.String, org.python.Object> kwargs = new HashMap<>();
```
Consider writing it like this instead so `kwargs` are aware of the types it contains.